File: GestionRestau/Models/Entities/Menu.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace BrasilBurger.Web.Models.Entities
{
    [Table("menus")]
    public class Menu
    {
        [Key]
        [Column("id_menu")]
        public int IdMenu { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("nom")]
        public string Nom { get; set; } = string.Empty;

        [MaxLength(255)]
        [Column("image")]
        public string? Image { get; set; }

        [Column("actif")]
        public bool Actif { get; set; } = true;

        // Navigation properties
        public virtual ICollection<MenuBurger> MenuBurgers { get; set; } = new List<MenuBurger>();

        // Propriété calculée pour le prix (somme des burgers)
        [NotMapped]
        public decimal Prix => MenuBurgers?.Sum(mb => mb.Burger?.Prix ?? 0) ?? 0;
    }
}

File: Java/src/models/Menu.java
package models;

public class Menu {
    private int idMenu;
    private String nom;
    private boolean actif;

    public Menu() {}

    public Menu(int idMenu, String nom, boolean actif) {
        this.idMenu = idMenu;
        this.nom = nom;
        this.actif = actif;
    }

    public int getIdMenu() { return idMenu; }
    public void setIdMenu(int idMenu) { this.idMenu = idMenu; }
    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }
    public boolean isActif() { return actif; }
    public void setActif(boolean actif) { this.actif = actif; }
}
